test/integration/scheduler/predicates_test.go

```diff
@@ -870,3 +870,68 @@ func TestInterPodAffinity(t *testing.T) {
 		}
 	}
 }
+
+// TestNodePIDPressue verifies that scheduler's CheckNodePIDPressurePredicate predicate
+// functions works correctly.
+func TestNodePIDPressue(t *testing.T) {
+	context := initTest(t, "node-pid-pressure")
+	defer cleanupTest(t, context)
+	// Add a few nodes.
+	nodes, err := createNodes(context.clientSet, "testnode", nil, 2)
+	if err != nil {
+		t.Fatalf("Cannot create nodes: %v", err)
+	}
+
+	cs := context.clientSet
+
+	// Adds PID pressure condition to the frist node.
+	nodes[0].Status.Conditions = []v1.NodeCondition{
+		{
+			Type:   v1.NodePIDPressure,
+			Status: v1.ConditionTrue,
+		},
+	}
+
+	// So the second node is the target node.
+	targetNode := nodes[1].Name
+
+	// Creats test pod.
+	testPod := &v1.Pod{
```

Comment:
In order to ensure that the pod is not randomly assigned to the second node, I would add `nodeSelector` to the spec to choose the first node and would check that the pod is marked unschedulable.

Reply:
Good point :)

Reply:
done